src/charm.py

```diff
@@ -202,85 +225,102 @@ def reconfigure_nats(self):
-                'tls_cert_path': self.TLS_CERT_PATH,
-                'verify_tls_clients': self.model.config['verify-tls-clients'],
-                'map_tls_clients': self.model.config['map-tls-clients'],
-                'tls_ca_cert_path': self.TLS_CA_CERT_PATH,
-            })
-            self.client.set_tls_ca(
-                self.ca_client.ca_certificate.public_bytes(encoding=serialization.Encoding.PEM).decode('utf-8'))
+            ctxt.update(
+                {
+                    "use_tls": True,
+                    "tls_key_path": self.TLS_KEY_PATH,
+                    "tls_cert_path": self.TLS_CERT_PATH,
+                    "verify_tls_clients": self.model.config["verify-tls-clients"],
+                    "map_tls_clients": self.model.config["map-tls-clients"],
+                    "tls_ca_cert_path": self.TLS_CA_CERT_PATH,
+                }
+            )
+            self.client._set_tls_ca(
+                self.ca_client.ca_certificate.public_bytes(
+                    encoding=serialization.Encoding.PEM
+                ).decode("utf-8")
+            )
 
         if self.nrpe_client.is_available:
             check_name = "check_{}".format(self.model.unit.name.replace("/", "_"))
-            self.nrpe_client.add_check(command=[
-                '/usr/lib/nagios/plugins/check_tcp',
-                '-H', str(self.client.listen_address),
-                '-p', str(self.model.config['client-port'])
-            ], name=check_name)
+            self.nrpe_client.add_check(
+                command=[
+                    "/usr/lib/nagios/plugins/check_tcp",
+                    "-H",
+                    str(self.client.listen_address),
+                    "-p",
+                    str(self.model.config["client-port"]),
+                ],
+                name=check_name,
+            )
             self.nrpe_client.commit()
 
-        tenv = Environment(loader=FileSystemLoader('templates'))
-        template = tenv.get_template('nats.cfg.j2')
+        tenv = Environment(loader=FileSystemLoader("templates"))
+        template = tenv.get_template("nats.cfg.j2")
         rendered_content = template.render(ctxt)
-        content_hash = self.generate_content_hash(rendered_content)
+        content_hash = self._generate_content_hash(rendered_content)
         old_hash = self.state.nats_config_hash
         if old_hash != content_hash:
-            logging.info(f'Config has changed - re-rendering a template to {self.NATS_SERVER_CONFIG_PATH}')
+            logging.info(
+                f"Config has changed - re-rendering a template to {self.NATS_SERVER_CONFIG_PATH}"
+            )
             self.state.nats_config_hash = content_hash
             self.NATS_SERVER_CONFIG_PATH.write_text(rendered_content)
             if self.state.is_started:
-                subprocess.check_call(['systemctl', 'restart', self.NATS_SERVICE])
+                subprocess.check_call(["systemctl", "restart", self.NATS_SERVICE])
```

Comment:
There is a [systemd library](https://charmhub.io/operator-libs-linux/libraries/systemd) that can help with this.